src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,40 @@
 package mate.academy;
 
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private static final int TIMEOUT = 10;
+    private final CopyOnWriteArrayList<EventListener> listeners = new CopyOnWriteArrayList<>();
```

Comment:
Using CopyOnWriteArrayList can be quite expensive for write operations as it involves copying the entire underlying array. If write operations are frequent compared to reads, consider using a different thread-safe list implementation or managing synchronization manually.